mgr/ConsDefs.pm

```diff
@@ -468,7 +468,7 @@
 	$CERNLIB_CPPFLAGS .= " -DCERNLIB_GCC" . $CXX_MAJOR;
         print "CXX_VERSION : $CXX_VERSION MAJOR = $CXX_MAJOR MINOR = $CXX_MINOR\n";
 
-        $CXXFLAGS    = "$XMACHOPT -fPIC -pipe -Wall -Woverloaded-virtual";
+        $CXXFLAGS    = "$XMACHOPT -fPIC -pipe -Wall -Woverloaded-virtual -Werror";
```

Comment:
Can we *please* limit this to CI. Unconditional `-Werror` can be a huge pain for some of the end users.

Reply:
yes, we can

Reply:
So, for implementation, we could define a cons variable like `WARNINGS_AS_ERRORS` and make this line conditional on it. What do you think?

Reply:
For now, I tried to use the known to cons variable `EXTRA_CXXFLAGS`. So, for pull request builds we can run it as:
```
cons EXTRA_CXXFLAGS="-Werror"
```